tests/cases/fourslash/contextuallyTypedParameters.ts

```diff
@@ -0,0 +1,60 @@
+/// <reference path='fourslash.ts' />
+
+////declare function foo(cb: (this: any, x: number, y: string, z: boolean) => void): void;
+////
+////foo(function(this, a, ...args) {

```

Comment:
Why write this is a fourslash test? Wouldn't the type baselines provide enough information to tell what the contextual type was?

Reply:
The problem only occurred when we were hovering in to get the type -- type baselines work "top down" like the commandline compiler and do not observe the problematic behavior